calico_containers/pycalico/util.py

```diff
@@ -290,25 +351,55 @@ def validate_cidr_versions(cidrs, ip_version=None):
             if ip_version is None:
                 ip_version = network.version
             elif ip_version != network.version:
-                return False
-    except (AddrFormatError, ValueError):
+                err = ValueError("IP Version does not match CIDR.")
+                return err if return_error else False
+    except (AddrFormatError, ValueError) as exc:
         # Some versions of Netaddr have a bug causing them to return a
         # ValueError rather than an AddrFormatError, so catch both.
-        return False
-    return True
+        err = ValueError("CIDR is invalid. " + str(exc).capitalize())
+        return err if return_error else False
+
+    return None if return_error else True
 
 
-def validate_ip(ip_addr, version):
+def validate_ip(ip_addr, version=None, return_error=False):
```

Comment:
Just remembered that I added `version=None` but didn't do anything with it. This validation function is so different than the rest, that I was hesitant  to touch it (none of the others have an assert). For simplicity sake, I'll probably just take out the `=None` addition, but there's really no reason to require a version when validating an IP unless you are explicitly checking for the version, which the name of this method doesn't imply.